src/modules/tracker.js

```diff
@@ -0,0 +1,1174 @@
+    headers['Accept-Language'] = userParameters.acceptLanguage;
+  }
+
+  // Append referrer as 'Referer' if available
+  if (userParameters.referer && typeof userParameters.referer === 'string') {
+    headers.Referer = userParameters.referer;
+  }
+
+  if (method === 'GET') {
+    request = fetch(url, { headers });
+  }
+
+  if (method === 'POST') {
+    request = fetch(url, {
+      method,
+      body: JSON.stringify(body),
+      mode: 'cors',
+      headers: {
+        ...headers,
+        'Content-Type': 'text/plain',
+      },
+    });
+  }
+
+  if (request) {
+    const instance = this;
+
+    request.then((response) => {
+      // Request was successful, and returned a 2XX status code
+      if (response.ok) {
+        instance.eventemitter.emit('success', {
+          url,
+          method,
+          message: 'ok',
+        });
+      }
+
+      // Request was successful, but returned a non-2XX status code
+      else {
+        response.json().then((json) => {
+          instance.eventemitter.emit('error', {
+            url,
+            method,
+            message: json && json.message,
+          });
+        }).catch((error) => {
+          instance.eventemitter.emit('error', {
+            url,
+            method,
+            message: error.type,
+          });
+        });
+      }
+    }).catch((error) => {
+      instance.eventemitter.emit('error', {
+        url,
+        method,
+        message: error.toString(),
+      });
+    });
+  }
+}
+
+/**
+ * Interface to tracking related API calls
+ *
+ * @module tracker
+ * @inner
+ * @returns {object}
+ */
+class Tracker {
+  constructor(options) {
+    this.options = options || {};
+    this.eventemitter = new EventEmitter();
+  }
+
+  /**
+   * Send session start event to API
+   *
+   * @function trackSessionStart
+   * @param {object} [userParameters] - Parameters relevant to the user request
+   * @param {number} [userParameters.sessionId] - Session ID, utilized to personalize results
+   * @param {number} [userParameters.clientId] - Client ID, utilized to personalize results
+   * @param {object} [userParameters.userId] - User ID, utilized to personalize results
+   * @param {string} [userParameters.segments] - User segments
+   * @param {string} [userParameters.testCells] - User test cells
+   * @param {string} [userParameters.originReferrer] - Client page URL (including path)
+   * @param {string} [userParameters.referer] - Client page URL (including path)
+   * @param {string} [userParameters.userIp] - Client user IP
+   * @param {string} [userParameters.userAgent] - Client user agent
+   * @param {string} [userParameters.acceptLanguage] - Client accept language
+   * @returns {(true|Error)}
+   */
+  trackSessionStart(userParameters) {
+    const url = `${this.options.serviceUrl}/behavior?`;
+    const queryParams = { action: 'session_start' };
+
+    send.call(
+      this,
+      `${url}${applyParamsAsString(queryParams, userParameters, this.options)}`,
```

Comment:
Can we pull this out into a `requestURL` variable like we do for the v2 endpoints and similarly do this for the other v1 endpoint calls?

Reply:
Sure can!